tests/codegen/checked_ilog.rs

```diff
@@ -1,11 +1,23 @@
 //@ compile-flags: -O
+//@ revisions: others riscv64 loongarch64
+
+//@[others] ignore-riscv64
+//@[others] ignore-loongarch64
+//@[riscv64] only-riscv64
+//@[riscv64] compile-flags: --target riscv64gc-unknown-linux-gnu
+//@[riscv64] needs-llvm-components: riscv
+//@[loongarch64] only-loongarch64
+//@[loongarch64] compile-flags: --target loongarch64-unknown-linux-gnu
+//@[loongarch64] needs-llvm-components: loongarch
 
 #![crate_type = "lib"]
 
 // Ensure that when val < base, we do not divide or multiply.
 
 // CHECK-LABEL: @checked_ilog
-// CHECK-SAME: (i16 noundef %val, i16 noundef %base)
+// others-SAME:      (i16 noundef %val, i16 noundef %base)
+// riscv64-SAME:     (i16 noundef zeroext %val, i16 noundef zeroext %base)
+// loongarch64-SAME: (i16 noundef zeroext %val, i16 noundef zeroext %base)
```

Comment:
like so?
```suggestion
// CHECK-SAME: (i16 noundef {{.*}} %val, i16 noundef {{.*}} %base)
```

or without the noundef? it's not clear to me if checking for that attr is useful for this test or not, honestly.
```suggestion
// CHECK-SAME: (i16 {{.*}} %val, i16 {{.*}} %base)
```

Reply:
It's fine to add the noundef into the wildcard as well, it's not relevant to the purpose of the test.

Reply:
Done. Since space matters in Regex matching, I had to use `ty{{( .+)?}} %var` instead of `ty {{.*}} %var`.

Reply:
Also added a separated test file `tests/codegen/rust-abi-arch-specific-adjustment.rs` specifically for the changes in this PR.